### PR TITLE
PLAT-670 Kafka dependent packages to use new Kafka

### PR DESCRIPTION
**Pull request by Mark-Labuschagne**
WIP, tasks to be complete (done for single-node and cluster):

- [x] Test with ethiopia-on-platform
- [x] Test with cares-on-platform
- [x] Test with disi-on-platform
- [x] Merge https://github.com/jembi/kafka-unbundler-consumer/pull/8
- [x] Merge https://github.com/jembi/kafka-mapper-consumer/pull/19

### CODE REVIEW

File: client-registry-jempi/docker-compose.kafdrop-dev.yml [hunk 1]
```diff
@@ -1,8 +0,0 @@
-version: '3.9'
-
-services:
-  jempi-kafdrop:
```
Comment (nour-borgi) [inline client-registry-jempi/docker-compose.kafdrop-dev.yml:4]: we need to remove the config importer in the importer folder as well (kafka.js and docker-compose.config.yml)

File: client-registry-jempi/package-metadata.json [hunk 2]
```diff
@@ -6,12 +6,6 @@
   "version": "0.0.1",
   "dependencies": ["interoperability-layer-openhim"],
   "environmentVariables": {
-    "JEMPI_KAFKA_01_MEMORY_LIMIT": "3G",
-    "JEMPI_KAFKA_01_MEMORY_RESERVE": "500M",
-    "JEMPI_KAFKA_02_MEMORY_LIMIT": "3G",
-    "JEMPI_KAFKA_02_MEMORY_RESERVE": "500M",
-    "JEMPI_KAFKA_03_MEMORY_LIMIT": "3G",
-    "JEMPI_KAFKA_03_MEMORY_RESERVE": "500M",
     "JEMPI_KAFDROP_CPU_LIMIT": "0",
```
Comment (nour-borgi) [inline client-registry-jempi/package-metadata.json:9]: we don't need kafdrop cpu and memory limits anymore

File: client-registry-jempi/package-metadata.json [hunk 3]
```diff
@@ -41,7 +35,6 @@
     "JEMPI_API_MEMORY_LIMIT": "3G",
     "JEMPI_API_MEMORY_RESERVE": "500M",
     "JEMPI_KAFKA_TOPICS": "JeMPI-async-preprocessor,JeMPI-patient-controller,JeMPI-patient-em,JeMPI-patient-linker,JeMPI-mu-linker,JeMPI-notifications",
```
Comment (nour-borgi) [inline client-registry-jempi/package-metadata.json:37]: I think this is something we need to care about, maybe we can put all these topics along side with the other topics in kafka folder, and they can be overwritten according to each implementation ?

File: client-registry-jempi/swarm.sh [hunk 4]
```diff
@@ -25,18 +24,12 @@ function init_vars() {
...
-    KAFKA_SERVICES=(
-      "${KAFKA_SERVICES[@]}"
-      "jempi-kafka-0$i"
-    )
     DGRAPH_SERVICES=(
       "${DGRAPH_SERVICES[@]}"
       "jempi-alpha-0$i"
     )
     VOLUME_NAMES=(
       "${VOLUME_NAMES[@]}"
```
Comment (nour-borgi) [inline client-registry-jempi/swarm.sh:32]: this can be removed also

File: client-registry-jempi/swarm.sh [hunk 5]
```diff
@@ -50,7 +43,6 @@ function init_vars() {
   )
 
   SERVICE_NAMES=(
-    "${KAFKA_SERVICES[@]}"
     "${DGRAPH_SERVICES[@]}"
     "${COMBINED_SERVICES[@]}"
     "jempi-kafdrop"
```
Comment (nour-borgi) [inline client-registry-jempi/swarm.sh:48]: kafdrop can be removed also

File: client-registry-jempi/swarm.sh [hunk 6]
```diff
@@ -102,14 +91,7 @@ function initialize_package() {
...
   (
-    log info "Deploy Kafka"
-    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.kafka.yml"
-    docker::deploy_sanity "${KAFKA_SERVICES[@]}"
-
-    log info "Deploy Kafdrop"
-    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.kafdrop.yml" "$kafdrop_dev_compose_param"
-    docker::deploy_sanity "jempi-kafdrop"
-
+    log info "Importing JeMPI Kafka topics"
```
Comment (nour-borgi) [inline client-registry-jempi/swarm.sh:94]: The env vars in docker compose config importer are still configured to "jempi-kafdrop', kafdrop host should be "kafdrop" right ?, also I think we need to add message-bus-kafka as a dependency to this package

File: data-mapper-logstash/docker-compose.yml [hunk 7]
```diff
@@ -13,6 +13,8 @@ services:
       ES_HOSTS: ${ES_HOSTS}
       ES_ELASTIC: ${ES_ELASTIC}
       LS_JAVA_OPTS: ${LS_JAVA_OPTS}
+      # Kafka as an input (required in the implementations)
+      KAKFA_HOSTS: ${KAKFA_HOSTS}
```
Comment (michaelloosen) [inline data-mapper-logstash/docker-compose.yml:17]: KAKFA_HOSTS should be KAFKA_HOSTS

**Review by michaelloosen — CHANGES_REQUESTED** [review]

File: data-mapper-logstash/docker-compose.yml (hunk 7 above)
Comment (nour-borgi) [reply data-mapper-logstash/docker-compose.yml]: cool

**Pull request merged**